### PR TITLE
Move methods only used in its own class to private (II)

### PR DESCRIPTION
**Pull request by eduardoj**
There is no need to make public all the methods of a class, part II.

Continue the work started in #7980.

### CODE REVIEW

**Comment by eduardoj** [comment]
> I normally think it's good, however i'm just wondering if all those methods are being indirectly tested, specially the one that you removed the test case because it was moved to private. Did you check the spec coverages if those methods are covered indirectly by tests @eduardoj ?

I only removed those lines of the test because I removed a method (`get_html`), that was not used.

In the rest of the cases, moving methods to private shouldn't affect the coverage results.

**Comment by vpereira** [comment]
> I only removed those lines of the test because I removed a method (`get_html`), that was not used.
> 
> In the rest of the cases, moving methods to private shouldn't affect the coverage results.

Yes, sorry if i wasn't clear. I'm not saying that this PR affects the coverage, I'm wondering that you were able to move those methods around without change, or delete, any test, which means that those methods were not directly tested  and we don't know if they are indirectly tested too.

File: src/api/app/models/buildresult.rb [hunk 1]
```diff
@@ -98,4 +92,12 @@ def self.summary(project_name)
 
     local_build_results
   end
+
+  private
+
+  def code2index(code)
```
Comment (vpereira) [inline src/api/app/models/buildresult.rb]: is that code being used anywhere?

Comment (eduardoj) [reply src/api/app/models/buildresult.rb]: Yep, in [this line](https://github.com/openSUSE/open-build-service/blob/25aafabde00ca52eeabde5c756a2628535da42ca/src/api/app/models/buildresult.rb#L94).

Comment (vpereira) [reply src/api/app/models/buildresult.rb]: argh, you are right my grep gave me the following output:

```
src/api/app/models/buildresult.rb
60:  def self.code2index(code)
94:      build.summary.sort! { |a, b| code2index(a.code) <=> code2index(b.code) }
```

and for me it looked like it was being used in the same method, but it's not!

**Pull request merged**